src/matrix/kaldi-matrix.cc

```diff
@@ -1437,9 +1439,11 @@ void Matrix<Real>::Read(std::istream & is, bool binary, bool add) {
     // Note, we never leave the while () loop before this
     // line (we return from it.)
  cleanup: // We only reach here in case of error in the while loop above.
-    delete cur_row;
+    if(cur_row != NULL)
```

Comment:
Just FYI, the check for NULL is not necessary (http://stackoverflow.com/questions/4190703/is-it-safe-to-delete-a-null-pointer).